data_model/model.py

```diff
@@ -98,37 +97,37 @@ def __init__(self):
 
     #    self.room_groups = {}
 
-    def add_room(self, room_id):
-        self.rooms[room_id] = room_id
+    def add_room(self, room):
+        self.rooms[room.id] = room
+        self.rooms.update({'id': room})
 
     # def add_room_group(self, room_group):
     #    self.room_groups[room_group.id] = room_group
 
     # def add_room_to_group(self, room, room_group):
-    #    self.room_groups[room_group.id] = []
     #    self.room_groups[room_group.id] += self.rooms[room.id]
 
-    def remove_room(self, room):
-        self.rooms.pop(room.id, None)
+    def remove_room(self, room_id):
+        self.rooms.pop(room_id, None)
 
     def get_room_by_id(self, room_id):
         try:
             return self.rooms[room_id]
         except KeyError:
             print("Room %d not found" % room_id)
 
-    def get_devices_for_room(self, room):
-        return self.rooms[room.id]
+    def get_devices_for_room(self, devices, room_id):
+        for device in devices:
+            if device.room_id == room_id:
+                return device
```

Comment:
You need to return all devices, so build a list of all devices that match and return it in the end.